src/components/MultiHopTrade/components/TradeInput/components/ManualAddressEntry.tsx

```diff
@@ -0,0 +1,117 @@
+import { swappers } from 'state/slices/swappersSlice/swappersSlice'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+type ManualAddressEntryProps = {
+  setIsManualAddressEntryValidating: (isManualAddressEntryValidating: boolean) => void
+}
+
+export const ManualAddressEntry: FC<ManualAddressEntryProps> = ({
+  setIsManualAddressEntryValidating,
+}): JSX.Element | null => {
+  const dispatch = useAppDispatch()
+
+  const {
+    formState: { isValid: isFormValid },
+    trigger: formTrigger,
+    setValue: setFormValue,
+  } = useFormContext()
+  const translate = useTranslate()
+  const isYatFeatureEnabled = useFeatureFlag('Yat')
+
+  const wallet = useWallet().state.wallet
+  const { chainId: buyAssetChainId, assetId: buyAssetAssetId } = useAppSelector(selectBuyAsset)
+  const isYatSupportedByReceiveChain = buyAssetChainId === ethChainId // yat only supports eth mainnet
+  const isYatSupported = isYatFeatureEnabled && isYatSupportedByReceiveChain
+
+  const walletSupportsBuyAssetChain = walletSupportsChain({ chainId: buyAssetChainId, wallet })
+  const shouldShowManualReceiveAddressInput = !walletSupportsBuyAssetChain
+
+  const chainAdapterManager = getChainAdapterManager()
+  const buyAssetChainName = chainAdapterManager.get(buyAssetChainId)?.getDisplayName()
+  const manualReceiveAddress = useAppSelector(selectManualReceiveAddress)
+
+  // Trigger re-validation of the manually entered receive address
+  useEffect(() => {
+    formTrigger(SendFormFields.Input)
+  }, [formTrigger, shouldShowManualReceiveAddressInput])
+
+  // Reset the manual address input state when the user changes the buy asset
+  useEffect(() => {
+    dispatch(swappers.actions.setManualReceiveAddress(undefined))
+  }, [buyAssetAssetId, dispatch])
+
+  useEffect(() => {
+    setFormValue(SendFormFields.Input, manualReceiveAddress ?? '')
+  }, [dispatch, manualReceiveAddress, setFormValue])
+
+  // For safety, ensure we never have a receive address in the store if the form is invalid
+  useEffect(() => {
+    !isFormValid && dispatch(swappers.actions.setManualReceiveAddress(undefined))
+  }, [isFormValid, dispatch])
+
+  const ManualReceiveAddressEntry: JSX.Element = useMemo(() => {
+    return (
+      <FormControl>
+        <FormLabel color='white.500' w='full' fontWeight='bold'>
+          {translate('trade.receiveAddress')}
+        </FormLabel>
+        <FormLabel color='yellow.400'>
+          {translate('trade.receiveAddressDescription', { chainName: buyAssetChainName })}
+        </FormLabel>
+        <AddressInput
+          rules={{
+            required: true,
+            validate: {
+              validateAddress: async (rawInput: string) => {
+                dispatch(swappers.actions.setManualReceiveAddress(undefined))
+                const value = rawInput.trim() // trim leading/trailing spaces
+                setIsManualAddressEntryValidating(true)
```

Comment:
Not introduced in this PR (this exists in prod too) but using boolean flag combined with async function introduces a race condition where the first invocation can set the flag to false while the second invocation is still running. 

To resolve this we should be using the provided `useFormState` hook that exposes internal `isValidating` flag so we don't have a race condition:
https://react-hook-form.com/docs/useformstate

Reply:
Also we should try to eliminate this callback to pass up the form state, perhaps add it to the slice?

Reply:
so in this case I'd add a useEffect to set the address validation state into the slice and delete the callback `setIsManualAddressEntryValidating` and have the parent component select it from the slice

Reply:
Good call. Fixed up in https://github.com/shapeshift/web/pull/4906/commits/a69f7bbdfbcfeca1ac22fa907abfc2ad69ab3651.